ports/nrf/mphalport.c

```diff
@@ -35,6 +35,120 @@
 #include "nrfx_errors.h"
 #include "nrfx_config.h"
 
+#if MICROPY_PY_TIME_TICKS
+#include "nrfx_rtc.h"
+#endif
+
+
+#if MICROPY_PY_TIME_TICKS
+// use RTC1 for time ticks generation
+// with 32kHz tick resolution and overflow
+// handling in irq
+
+nrfx_rtc_t rtc1 = NRFX_RTC_INSTANCE(1);
+
+volatile mp_uint_t rtc_overflows = 0;
+
+const nrfx_rtc_config_t rtc_config_time_ticks = {
+    .prescaler    = 0,
+    .reliable     = 0,
+    .tick_latency = 0,
+    #ifdef NRF51
+    .interrupt_priority = 1,
+    #else
+    .interrupt_priority = 3,
+    #endif
+};
+
+STATIC void rtc_irq_time(nrfx_rtc_int_type_t event) {
+    // irq handler for overflow
+    if (event == NRFX_RTC_INT_OVERFLOW) {
+        rtc_overflows += 1;
+    }
```

Comment:
If you add the following snippet, the CC will be periodic: 
```
if (event == NRFX_RTC_INT_COMPARE0) {
        uint32_t new_cc = nrfx_rtc_counter_get(&rtc1) + 30;
        nrfx_rtc_cc_set(&rtc1, 0, new_cc, true);
}
```

Reply:
Yes I did that just 30min ago, sorry ;)

Reply:
and 33 as increase to be a bit more precise with 1msec.